<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>835</width>
    <height>735</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* General Styles */
QWidget {
    background-color: #FBF9F1; /* Soft background */
    color: #333; /* Dark text for contrast */
    font-family: Arial, sans-serif;
}

/* Buttons */
QPushButton {
    background-color: #92C7CF; /* Soft blue */
    color: white;
    border-radius: 6px;
    padding: 6px 12px;
    border: none;
}
QPushButton:hover {
    background-color: #AAD7D9; /* Lighter hover */
}
QPushButton:pressed {
    background-color: #5E9EA0; /* Darker press effect */
}

/* Graph Viewer */
QGraphicsView {
    background-color: white;
    border: 1px solid #E5E1DA;
    border-radius: 6px;
}

/* Spin Boxes */
QSpinBox, QDoubleSpinBox {
    background-color: white;
    border: 1px solid #E5E1DA;
    padding: 4px;
    border-radius: 4px;
}
QSpinBox::up-button, QDoubleSpinBox::up-button,
QSpinBox::down-button, QDoubleSpinBox::down-button {
    width: 16px;
    background-color: #E5E1DA;
}
QSpinBox::up-button:hover, QDoubleSpinBox::up-button:hover,
QSpinBox::down-button:hover, QDoubleSpinBox::down-button:hover {
    background-color: #AAD7D9;
}

/* Labels */
QLabel {
    color: #333;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QSpinBox" name="nodesSpinBox">
    <property name="geometry">
     <rect>
      <x>255</x>
      <y>90</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="densityDoubleSpinBox">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>90</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="generateButton">
    <property name="geometry">
     <rect>
      <x>205</x>
      <y>140</y>
      <width>226</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate Graph</string>
    </property>
   </widget>
   <widget class="QPushButton" name="degreeButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>75</y>
      <width>176</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Show Node Degree</string>
    </property>
   </widget>
   <widget class="QPushButton" name="maxDegreeButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>45</y>
      <width>176</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string> Maximum Degree</string>
    </property>
   </widget>
   <widget class="QPushButton" name="minDegreeButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>105</y>
      <width>176</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Minimum Degree</string>
    </property>
   </widget>
   <widget class="QPushButton" name="neighborsButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>135</y>
      <width>176</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Find Neighbors</string>
    </property>
   </widget>
   <widget class="QPushButton" name="matrixButton">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>165</y>
      <width>176</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Display Adjacency Matrix</string>
    </property>
   </widget>
   <widget class="GraphWidget" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>215</y>
      <width>621</width>
      <height>466</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>95</y>
      <width>106</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of Nodes</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>95</y>
      <width>86</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Graph Density</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>235</x>
      <y>10</y>
      <width>256</width>
      <height>66</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <italic>true</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>GRAPH VISULIZER </string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>835</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GraphWidget</class>
   <extends>QGraphicsView</extends>
   <header>graphwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
